concent_api/core/views.py

```diff
@@ -93,6 +94,14 @@ def receive(request, _message):
                     decoded_message_data,
                     last_delivered_message_status,
                 )
+        acceptance_deadline = settings.SUBTASK_VERIFICATION_TIME + settings.FORCE_ACCEPTANCE_TIME + settings.CONCENT_MESSAGING_TIME
+        if (
+            last_delivered_message_status.message.type                           == message.concents.ForceSubtaskResults.TYPE and
+            last_delivered_message_status.message.auth.provider_public_key_bytes == client_public_key and
+            current_time > decoded_message_data.ack_report_computed_task.timestamp + acceptance_deadline
+        ):
+            make_forced_payment('provider', 'requestor')
+            return handle_receive_force_subtask_results(request, decoded_message_data, last_delivered_message_status.message.auth.provider_public_key_bytes, ReceiveStatus)
```

Comment:
In this usage the view is called by provider, but you are storing client_public_key as requestor key in 543.

Reply:
:+1: